wordpress-mobile/WordPress-iOS
# [JP Plugin] Tweak the Jetpack Remote Install UI

Resolves #20204 

This PR refactors the Jetpack Remote Install UI a bit to implement the improvements detailed in #20204, along with some accessibility fixes. 

Here's a preview video:

> **Note**: For the error flow video, perhaps the duration was too short so I couldn't see the error screen when playing the video from start to finish. But, it appears for me when you click on the video seeker thingy and manually drag it to the end.

Success flow | Error flow
-|-
<video src="https://user-images.githubusercontent.com/1299411/221668409-0f563f2b-76d6-4ad0-b2ad-6c92cc5b7a78.mp4"> | <video src="https://user-images.githubusercontent.com/1299411/221668429-4dd705a3-31b2-4cb3-aed8-4adad264f951.mp4">

For @osullivanchris, this is not testable yet from the Installable Build since you'll still need to make some code changes locally to enable the whole thing. For now, please verify visually through the recordings I've attached above. Thanks!

## To test

- Enable the `jetpackIndividualPluginSupport` feature flag.
- Prepare a self-hosted site with an individual Jetpack plugin and have it connected to your WordPress.com account.
- Run the Jetpack app and log in with your WordPress.com account.
- Switch to the self-hosted site.
- When the overlay appears, tap the "Install the full plugin" button.
- 🔎 Verify that the screen is displayed according to the new design.
- To verify the error path first, turn on airplane mode / turn off your connection. 
- Tap "Continue".
- The loading state only appears for a short bit, but you can verify this later. 
- 🔎  The error screen should appear. Verify that the screen is displayed according to the new design.
- Now, turn off airplane mode and tap "Retry".
- 🔎 Verify that the button shows a loading state.
- 🔎 Verify that tapping on the button does nothing.
- The success screen should appear after the installation succeeded.
- 🔎 Verify that the screen is displayed according to the new design.

## Regression Notes
1. Potential unintended areas of impact
Should be none.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:
